# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.cpp
# Commit: 584beaae1d20
# Full Hash: 584beaae1d20fcf1f066f802e6412673028fa4e5
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 3
# Overlapping Files: js/loader/ModuleLoadRequest.h, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp
# Description:
#   Bug 1968890 - Part 5: Add ModuleLoadRequest::SetErroredLoadingImports(). r=jonco
# ==============================================================================

diff -r e2701bd43ce2 -r 584beaae1d20 js/loader/ModuleLoaderBase.cpp
--- a/js/loader/ModuleLoaderBase.cpp	Fri Oct 03 20:53:29 2025 +0000
+++ b/js/loader/ModuleLoaderBase.cpp	Fri Oct 03 20:53:29 2025 +0000
@@ -1325,6 +1325,7 @@
                                 JSMSG_DYNAMIC_IMPORT_FAILED, url.get());
       FinishLoadingImportedModuleFailedWithPendingException(aCx, payload);
     }
+    aRequest->SetErroredLoadingImports();
   } else if (moduleScript && !error.isUndefined()) {
     LOG(
         ("ScriptLoadRequest (%p): LoadRequestedModules rejected: set error to "
